### DISCUSSION TITLE
Camera stream transfer to another pc over lan

### DISCUSSION THREAD

**Discussion by prakhar114** [eclipse-ecal/ecal#1243] [Q&A]
Hello, 

I want to send 5 camera streams(1920x1080) to another PC connected with rj45 cable. 

Requirements device1:

- The system should receive 5 camera buffer from a publisher(ecal IPC ) and then send the buffer to device2 using (TCP/UDP).

Requirement for device2:

- The system should receive the buffer from device1 and send it to other subscribers(ecal IPC)

I want to know how feasible it is to use such a system and whether I can maintain 20-30 fps, if yes, please guide me in the right direction to implement this using ecal.

**Comment by FlorianReimold** [comment]
Well, let's do the math.

Considering uncompressed, debayered images (3 byts per pixel), the size of an Image is:
$`(1920 * 1080) * 3 [bytes] = 6220800 [bytes] `$

Sending 5 of those with 30 FPS adds up to:
$`6220800 [bytes] * 5 * 30 \frac{1}{[sec]} = 933120000 [\frac{bytes}{sec}] = 889.892 [\frac{MiB}{sec}] = 6.952 [\frac{Gbit}{sec}] `$

Does your connection have that link speed? You will probably have to have 10 Gbit/s interfaces on both machines. If you plan on compressing the image, you can redo the math to calculate your required link speeds.

**Comment by prakhar114** [comment]
yes, I have a 10 Gbit/sec interface on both the devices

To create a pipeline, I was trying to run the sample codes to start a basic communication but it's not working.
 I have connected both computers through an RJ45 cable without a router or a switch in between, both the pc's are connected and can ping each other, but when I try to run the minimal_client_c and server code on both PCs, I get this error on the client side.

```
Calling service1:echo ..
Service/method not found :-(

```

I am very new to networking, so apologies in advance if it's a silly question.

**Comment by FlorianReimold** [comment]
You have probably already seen our online Getting Started guide:
https://eclipse-ecal.github.io/ecal/getting_started/cloud.html

You should start off following it. You will not need any eCAL Services for now, but it is a good idea to get them working as well. The eCAL TCP Layer is rather similar to the eCAL Services and if the Services don't work, then the eCAL TCP Layer will probably not work as well.

So:
1. Install eCAL on both of your machines (https://eclipse-ecal.github.io/ecal/getting_started/setup.html)
2. If you have a direct ethernet connection, assign an IP address to both of them
3. Test on each machine, if the eCAL person sample runs locally and shows communication (https://eclipse-ecal.github.io/ecal/getting_started/samples.html)
4. Configure the Multicast routes. This is important for both machines to find each other over network (https://eclipse-ecal.github.io/ecal/getting_started/cloud.html)
5. Check if the person sample can now communicate between your two machines. 
6. Also make sure that they can resolve each other's hostname to the IP address that you assigned them. As mentioned, this is important both for the eCAL Services, as well as for the eCAL TCP Layer (https://eclipse-ecal.github.io/ecal/getting_started/services.html)
7. Now try if the eCAL TCP sample can communicate between your two machines (ecal_sample_person_snd_tcp / ecal_sample_person_rec_tcp)

If everything works, you should be ready to send big data between your machines. You should use the C++ API and activate TCP for your publisher in order to force eCAL to use TCP.
